plugins/com.python.pydev.analysis/src/com/python/pydev/analysis/flake8/Flake8CodesConfigHandler.java

```diff
@@ -0,0 +1,205 @@
+        ret.put("warning", IMarker.SEVERITY_WARNING);
+        ret.put("info", IMarker.SEVERITY_INFO);
+        return ret;
+    }
+
+    private static Optional<String> checkSeverity(String codeRep, JsonValue severity) {
+        if (severity.isNumber()) {
+            Collection<Integer> possibleNumbers = POSSIBLE_SEVERITIES.values();
+            Optional<String> errorMessage = Optional.of(codeRep + " value does not match any string or "
+                    + possibleNumbers.toString() + " integer error values");
+            try {
+                if (!possibleNumbers.contains(severity.asInt())) {
+                    return errorMessage;
+                }
+            } catch (Exception e) {
+                return errorMessage;
+            }
+        } else if (severity.isString()) {
+            Set<String> possibleStrings = POSSIBLE_SEVERITIES.keySet();
+            String severityStr = severity.asString().trim();
+            if (!possibleStrings.contains(severityStr)) {
+                return Optional.of(codeRep + " value does not match any integer or "
+                        + possibleStrings.toString() + " string error values");
+            }
+        } else {
+            return Optional.of(codeRep + " value is neither a string or an integer");
+        }
+        return Optional.empty();
+    }
+
+    public static boolean checkRangeOverlap(Tuple<Integer, Integer> compareRange, Tuple<Integer, Integer> baseRange) {
+        int c1 = compareRange.o1;
+        int c2 = compareRange.o2;
+        int b1 = baseRange.o1;
+        int b2 = baseRange.o2;
+
+        if (b1 == c1 || b2 == c2) {
+            return true;
+        }
+        if ((c1 <= b2 || b2 == -1) && (b1 <= c2 || c2 == -1)) {
+            return true;
+        }
+        return false;
+    }
+
+    private final static Tuple<Integer, Integer> getRangeFromIdentifier(String identifier) {
+        if (identifier.contains("[")) {
+            String valuesStr = identifier.substring(1, identifier.length() - 1);
+            List<String> values = StringUtils.split(valuesStr, ',');
+            int firstValue = Integer.parseInt(values.get(0).trim());
+            int secondValue = Integer.parseInt(values.get(1).trim());
+            return new Tuple<Integer, Integer>(firstValue, secondValue);
+        } else if (identifier.isBlank()) {
+            return new Tuple<Integer, Integer>(-1, -1);
+        } else {
+            int value = Integer.parseInt(identifier);
+            return new Tuple<Integer, Integer>(value, value);
+        }
+    }
+
+    public static Map<String, Tuple<Set<Tuple<Integer, Integer>>, Integer>> getCodeSeveritiesFromConfig(
+            IAdaptable projectAdaptable) {
+        Map<String, Tuple<Set<Tuple<Integer, Integer>>, Integer>> ret = new HashMap<String, Tuple<Set<Tuple<Integer, Integer>>, Integer>>();
+        String str = Flake8Preferences.getCodesConfig(projectAdaptable);
+        JsonObject jsonObject = JsonObject.readFrom(str);
```

Comment:
We should handle errors if the json cannot be interpreted (if an error happens, just log the error and return and empty HashMap).